tests/specs/functional/lemmas-spec.k

```diff
@@ -27,14 +27,16 @@ module LEMMAS-SPEC
  // -----------------------
 
     rule <k> runLemma ( #lookupMemory((KEY |-> 33) (KEY' |-> 728) (KEY'' |-> (pow256 +Int 5)) (KEY''' |-> "hello"), KEY)    ) => doneLemma ( 33  ) ... </k>
-    // rule <k> runLemma ( #lookupMemory((KEY |-> 33) (KEY' |-> 728) (KEY'' |-> (pow256 +Int 5)) (KEY''' |-> "hello"), KEY')   ) => doneLemma ( 216 ) ... </k>
-    // rule <k> runLemma ( #lookupMemory((KEY |-> 33) (KEY' |-> 728) (KEY'' |-> (pow256 +Int 5)) (KEY''' |-> "hello"), KEY'')  ) => doneLemma ( 5   ) ... </k>
-    // rule <k> runLemma ( #lookupMemory((KEY |-> 33) (KEY' |-> 728) (KEY'' |-> (pow256 +Int 5)) (KEY''' |-> "hello"), KEY''') ) => doneLemma ( 0   ) ... </k>
+    rule <k> runLemma ( #lookupMemory((KEY |-> 33) (KEY' |-> 728) (KEY'' |-> (pow256 +Int 5)) (KEY''' |-> "hello"), KEY')   ) => doneLemma ( 216 ) ... </k>
+    rule <k> runLemma ( #lookupMemory((KEY |-> 33) (KEY' |-> 728) (KEY'' |-> (pow256 +Int 5)) (KEY''' |-> "hello"), KEY'')  ) => doneLemma ( 5   ) ... </k>
+    rule <k> runLemma ( #lookupMemory((KEY |-> 33) (KEY' |-> 728) (KEY'' |-> (pow256 +Int 5)) (KEY''' |-> "hello"), KEY''') ) => doneLemma ( 0   ) ... </k>
+    rule <k> runLemma ( #lookupMemory((KEY |-> 33), KEY') ) => doneLemma ( 0 ) ... </k> requires notBool KEY' in_keys(KEY |-> 33)
```

Comment:
@denis-bogdanas Can we make this more uniform with others?  Like,
```suggestion
    rule <k> runLemma ( #lookupMemory((KEY |-> 33) (KEY' |-> 728) (KEY'' |-> (pow256 +Int 5)) (KEY''' |-> "hello"), KEY'''') ) => doneLemma ( 0 ) ... </k> requires KEY'''' =/=K KEY andBool KEY'''' =/=K KEY'' andBool KEY'''' =/=K KEY'''
```

Reply:
The inequalities `KEY'''' =/=K KEY` won't help evaluating `in_keys`. Having `in_keys(the whole 4-key map)` looks silly. I like this version better. Sadly it doesn't work and there is some other issue on Java backend, will look tomorrow probably.

Reply:
I see. Never mind then.